crates/rnote-ui/src/dialogs/mod.rs

```diff
@@ -552,6 +553,38 @@ pub(crate) async fn dialog_edit_selected_workspace(appwindow: &RnAppWindow) {
     dialog.present(appwindow);
 }
 
+pub(crate) async fn dialog_trash_file(appwindow: &RnAppWindow, filerow: &RnFileRow) -> () {
```

Comment:
Instead of passing the filerow, directly pass the file. Then the implementation of the dialog is decoupled from the filerow. You'll need to pass the file as a `@strong` reference to all closures then

Reply:
Done, thanks for the tip. I initially tried getting it to work like this, but had issues with the weak reference.